pkg/forwarder/forwarder.go

```diff
@@ -355,6 +359,10 @@ func (f *DefaultForwarder) createPriorityHTTPTransactions(endpoint endpoint, pay
 				t.Headers.Set(versionHTTPHeaderKey, version.AgentVersion)
 				t.Headers.Set(useragentHTTPHeaderKey, fmt.Sprintf("datadog-agent/%s", version.AgentVersion))
 
+				if f.completionHandler != nil {
+					t.completionHandler = f.completionHandler
```

Comment:
this logic is pretty simple, but could you still add a test case to cover it in `forwarder_test.go`?

Reply:
Sure!

Reply:
@olivielpeau done! https://github.com/DataDog/datadog-agent/pull/6525/commits/3783bb8c148dae067d393a796dfb076ec6073091